### PR TITLE
feature : implement alias group/categories

### PR DESCRIPTION
**Pull request by virat-sr**
## PR Checklist
- [x] Follows single-purpose principle  
- [x] Tests pass locally (if applicable)
- [x] Documentation updated (if needed)

## What does this PR do?
Implements alias grouping functionality to allow users to organize aliases into logical groups like "git", "docker", "kubernetes" for better organization and management.

**Key Features Added:**
- Add `group` field to Alias model with proper storage/loading
- CLI group management commands (`alix group create`, `list`, `add`, `remove`, `delete`, `apply`)
- TUI group filtering with 'g' key binding
- Group column in TUI table display
- Bulk operations on groups (apply all aliases in group, delete group, reassign group)
- Group field in add/edit modals

## Related Issue
Fixes #(issue number) - Add alias grouping functionality

## Type of change
- [x] New feature (non-breaking)
- [ ] Bug fix (non-breaking)
- [ ] Configuration change
- [ ] Documentation update
- [ ] Setup/Infrastructure

### CLI Group Commands
```bash
# Create test aliases and groups
python -m alix add --name "gst" --command "git status" --description "Git status"
python -m alix add --name "dps" --command "docker ps" --description "Docker ps"
python -m alix group add git gst
python -m alix group add docker dps

# Test group operations
python -m alix group list                    # List all groups
python -m alix group apply git              # Apply all git aliases
python -m alix group delete docker          # Delete docker group
```

### TUI Group Functionality
```bash
# Launch TUI
python -m alix

# Test group filtering
# Press 'g' to cycle through groups: All Groups → git → docker → Ungrouped → All Groups
# Verify group column shows correct group names
# Test adding/editing aliases with group field
```

### Bulk Operations
```bash
# Test bulk apply
python -m alix group apply git              # Should apply all aliases in git group

# Test bulk delete with reassignment
python -m alix group delete git --reassign git-advanced  # Should move all git aliases to git-advanced group
```

### Edge Cases
- Test operations on non-existent groups
- Test operations on empty groups
- Test group filtering with no groups
- Test data persistence after restart

## Files Changed
- `models.py` - Added group field to Alias model and to_dict/from_dict methods
- `cli.py` - Added group subcommand with create, list, add, remove, delete, apply commands
- `tui.py` - Added group column, group filtering, group field in modals
- `storage.py` - Added group-specific storage methods

## Breaking Changes
None - this is a non-breaking feature addition.

### CODE REVIEW

File: alix/cli.py [hunk 1]
```diff
@@ -280,6 +282,247 @@ def list_aliases():
...
+        for alias in group_aliases:
+            alias.group = None
+            storage.aliases[alias.name] = alias
+        storage.save()
+        console.print(f"[green]✓[/] Removed group '{group_name}' from {len(group_aliases)} aliases")
+
+@group.command()
+@click.argument("group_name")
+@click.option("--apply", is_flag=True, help="Apply all aliases in group to shell")
+def apply(group_name, apply):
```
Comment (Valentin-v-Todorov) [inline alix/cli.py:430]: The "apply" command is defined TWICE! This will cause a runtime error. Remove the duplicate. ✌️

File: alix/cli.py [hunk 2]
```diff
@@ -280,6 +282,247 @@ def list_aliases():
...
+        if skipped_count > 0:
+            console.print(f"[yellow]⚠[/] Skipped {skipped_count} existing aliases")
+            
+    except Exception as e:
+        console.print(f"[red]✗[/] Failed to import: {e}")
+
+@group.command()
+@click.argument("group_name")
+@click.option("--apply", is_flag=True, help="Apply all aliases in group to shell")
+def apply(group_name, apply):
```
Comment (Valentin-v-Todorov) [inline alix/cli.py:498]: The duplicate

File: alix/cli.py (hunk 1 above)
Comment (virat-sr) [reply alix/cli.py]: Removed!

File: alix/cli.py (hunk 2 above)
Comment (virat-sr) [reply alix/cli.py:466]: I usually add new functions at the end rather than in the middle for better Git tracking. The earlier one was left in during development, but I forgot to remove it before raising the PR.

File: alix/tui.py [hunk 3]
```diff
@@ -696,4 +711,97 @@ def on_button_pressed(self, event: Button.Pressed) -> None:
...
+        
+        # Apply the filter
+        if selected_group == "All Groups":
+            self.refresh_table()
+            self.notify("Showing all aliases", severity="information")
+        elif selected_group == "Ungrouped":
+            self.refresh_table_by_group(None)
+            self.notify("Showing ungrouped aliases", severity="information")
+        else:
+            self.refresh_table_by_group(selected_group)
```
[the remark this answers is not in the document]
Comment (virat-sr) [reply alix/tui.py]: I had initially written my own refresh function but later realized that refresh_table already exists. I removed the function definition, but it seems I forgot to remove the function call.

**Comment by virat-sr** [comment]
I guess we need to discuss changes for tui.py

**Comment by virat-sr** [comment]
I fixed all merge conflicts as well. Please check

**Comment by Valentin-v-Todorov** [comment]
@virat-sr LGTM 

The TUI integration is well done 🥇 

Great job!
